src/main/java/de/neuland/jade4j/model/JadeModel.java

```diff
@@ -50,6 +57,16 @@ public MixinNode getMixin(String name) {
 		return mixins.get(name);
 	}
 
+	public void visit(Node node, JadeTemplate template) {
+		String visitId = node.getFileName()+":"+node.getLineNumber();
+		Integer ct = visited.getOrDefault(visitId, 0) +1;
+		if (ct > MAX_VISITED) {
+			String error = "Potential infinite loop: "+visitId+" has been visited more then "+MAX_VISITED+" times!";
+			throw new JadeCompilerException(node, template.getTemplateLoader(), error);
```

Comment:
Can we get the Template Name somehow and return it in the Exception message?

Reply:
`visitId` contains file path and line number